pr_review_queue.py

```diff
@@ -244,24 +245,65 @@ def create_pr_review_queue(pull_request_list):
     Return a filtered list of pull requests according to these criteria:
         1. CI is green
         2. Not a draft
-        3. No changes requested
-        4. No merge conflicts
+    The resulting pull requests are grouped into four sections:
+        1. ‘Needs reviewer’: ping author about finding a reviewer
+        2. ‘Needs changes’: ping author to do the changes
+        3. ‘Needs review’: ping assigned reviewer/s
+        4. ‘Needs conflict resolution’: ping author to rebase/fix conflicts
     """
-    pr_review_queue = []
-    i = 0
+    needs_reviewer = []
+    needs_changes = []
+    needs_review = []
+    needs_conflict_resolution = []
+
     for pull_request in pull_request_list:
+        # 1. Needs reviewer
         if (pull_request["status"] == "success" and
             pull_request["draft"] == False and
             pull_request["changes_requested"] == False and
+            pull_request["mergeable_state"] != "dirty" and
+            pull_request["requested_reviewers"] == []):
+            entry = (f"*{pull_request['repo']}*:"
+                     f" <{pull_request['html_url']}|{pull_request['title']}>"
```

Comment:
It would be super fancy if we could also detect Jira keys and split this into two links
pseudo code:

...
jira_key, separator, title_remainder = find_jira_key(pull_request['title'])
if not jira_key:
  entry += f" <{pull_request['html_url']}|{pull_request['title']}>"
else:
  entry += f" {generate_jira_link(jira_key)}{separator}<{pull_request['html_url']}|{title_remainder}>
  
where "separator" should be the things between the key and the title (e.g. ": ") so the two links are separated by the colon and the space....

def find_jira_key(title):
    ...TBD should only be one regex with three groups :-D

def generate_jira_link(jira_key):
    ...TBD maybe check if the URL returns 200 and don't do the link otherwise but that's really overkill I guess

Reply:
Great idea! I'll do that in a follow-up (created an issue)